File: glue/rust/examples/blink.rs
// Blinking LED example
use warduino::{delay, digital_write, pin_mode, PinMode, PinVoltage};

mod config;

#[no_mangle]
pub fn main() {
    let led: u32 = config::LED;
    pin_mode(led, PinMode::OUTPUT);

    let pause: u32 = 1000;
    loop {
        digital_write(led, PinVoltage::HIGH);
        delay(pause);
        digital_write(led, PinVoltage::LOW);
        delay(pause);
    }
}

File: glue/assemblyscript/examples/blink.ts
// Blinking LED example
import {pinMode, PinMode, PinVoltage, digitalWrite, delay} from "as-warduino/assembly";
import {config} from "./config";

export function main(): void {
    let led = config.led;
    pinMode(led, PinMode.OUTPUT);

    let pause: u32 = 1000;
    while (true) {
        digitalWrite(led, PinVoltage.HIGH);
        delay(pause);
        digitalWrite(led, PinVoltage.LOW);
        delay(pause);
    }
}
